src/Generic4626Wrapper.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import {IPool} from 'aave-v3-core/contracts/interfaces/IPool.sol';
+import {ICreditDelegationToken} from 'aave-v3-core/contracts/interfaces/ICreditDelegationToken.sol';
+import {IERC20} from 'aave-v3-core/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {IERC4626} from './dependencies/IERC4626.sol';
+import {IUSDS} from './dependencies/IUSDS.sol';
+import {BaseTokenWrapper} from './BaseTokenWrapper.sol';
+
+/**
+ * @title Generic4626Wrapper
+ * @author Aave
+ * @notice Generic contract to wrap an ERC20 to ERC4626 to on supply to Aave, or unwrap from ERC4626 to ERC20 on withdrawal
+ */
+contract Generic4626Wrapper is BaseTokenWrapper {
+  /**
+   * @dev Constructor
+   * @param tokenIn Address for the ERC20 token
+   * @param tokenOut Address for the ERC4626 token
+   * @param pool The address of the Aave Pool
+   * @param owner The address to transfer ownership to
+   */
+  constructor(
+    address tokenIn,
+    address tokenOut,
+    address pool,
+    address owner
+  ) BaseTokenWrapper(tokenIn, tokenOut, pool, owner) {
+    IERC20(tokenIn).approve(tokenOut, type(uint256).max);
+  }
+
+  // @inheritdoc BaseTokenWrapper
+  function borrowToken(uint256 amount, uint16 referralCode) external override {
+    _borrowToken(amount, msg.sender, referralCode);
+  }
+
+  /// @inheritdoc BaseTokenWrapper
+  function borrowTokenWithPermit(
```

Comment:
I d move this to the base contract

Reply:
Done [f8b02c0](https://github.com/aave/token-wrappers/commit/f8b02c016564751f9c40f3e236bd3bee738b94d5)